### PR TITLE
[Messenger] HandlerArgumentsStamp

### PR DESCRIPTION
**Pull request by enumag**
Refs
* https://github.com/symfony/symfony/pull/45418

### CODE REVIEW

**Comment by alexislefebvre** [comment]
Can you please add an example of code?

I think it also need a note to say that it was added since Symfony 6.2.

**Comment by OskarStark** [comment]
friendly ping @enumag

**Comment by enumag** [comment]
Code sample and version note added.

File: messenger.rst [hunk 1]
```diff
@@ -2413,6 +2413,62 @@ of the process. For each, the event class is the event name:
 * :class:`Symfony\\Component\\Messenger\\Event\\WorkerRunningEvent`
 * :class:`Symfony\\Component\\Messenger\\Event\\WorkerStoppedEvent`
 
+Additional Handler Arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It's possible to have messenger pass additional data to the message handler using the
+:class:`Symfony\\Component\\Messenger\\Stamp\\HandlerArgumentsStamp`. Add this stamp to
+the envelope in a middleware and fill it with any additional data you want to have
+available in the handler.
```
Comment (MrYamous) [inline messenger.rst]:
```suggestion
available in the handler::
```

File: messenger.rst [hunk 2]
```diff
@@ -2413,6 +2413,62 @@ of the process. For each, the event class is the event name:
...
+            return $stack->next()->handle($envelope, $stack);
+        }
+
+        private function resolveAdditionalArgument(object $message): mixed
+        {
+            // ...
+        }
+    }
+
+Then your handler will look like this:
```
Comment (MrYamous) [inline messenger.rst]:
```suggestion
Then your handler will look like this::
```

File: messenger.rst [hunk 3]
```diff
@@ -2413,6 +2413,62 @@ of the process. For each, the event class is the event name:
...
+    {
+        public function __invoke(SmsNotification $message, mixed $additionalArgument)
+        {
+            // ...
+        }
+    }
+
+.. versionadded:: 6.2
+
+    The HandlerArgumentsStamp was introduced in Symfony 6.2.
```
Comment (MrYamous) [inline messenger.rst]:
```suggestion
    The :class:`Symfony\\Component\\Messenger\\Stamp\\HandlerArgumentsStamp` was introduced in Symfony 6.2.
```

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
It took time, but here we go, this is in now. Thank you very much Jáchym.

**Pull request merged**